tests/test_opengraph.py

```diff
@@ -41,3 +41,215 @@ def test_open_graph_to_pattern() -> None:
     og_reconstructed = OpenGraph.from_pattern(pattern)
 
     assert og.isclose(og_reconstructed)
+
+
+# Tests composition of two graphs
+
+
+# Parallel composition
+def test_compose_1() -> None:
+    # Graph 1
+    # [1] -- (2)
+    #
+    # Graph 2 = Graph 1
+    #
+    # Mapping: 1 -> 100, 2 -> 200
+    #
+    # Expected graph
+    #  [1]  --  (2)
+    #
+    # [100] -- (200)
+
+    g: nx.Graph[int]
+    g = nx.Graph([(1, 2)])
+    inputs = [1]
+    outputs = [2]
+    meas = {i: Measurement(0, Plane.XY) for i in g.nodes - set(outputs)}
+    og_1 = OpenGraph(g, meas, inputs, outputs)
+
+    mapping = {1: 100, 2: 200}
+
+    og, mapping_complete = og_1.compose(og_1, mapping)
+
+    expected_graph: nx.Graph[int]
+    expected_graph = nx.Graph([(1, 2), (100, 200)])
+    assert nx.is_isomorphic(og.inside, expected_graph)
+    assert og.inputs == [1, 100]
+    assert og.outputs == [2, 200]
+
+    outputs_c = [i for i in og.inside.nodes if i not in og.outputs]
+    assert len(og.measurements) == len(outputs_c)
+    assert set(og.measurements) == set(outputs_c)
```

Comment:
Sorry, I missed this one!
```suggestion
    outputs_c = {i for i in og.inside.nodes if i not in og.outputs}
    assert og.measurements.keys() == outputs_c
```